oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AuthorizationCodeRequestAuthenticationProvider.java

```diff
@@ -295,26 +339,6 @@ private static OAuth2TokenContext createAuthorizationCodeTokenContext(
 		return tokenContextBuilder.build();
 	}
 
-	private static boolean requireAuthorizationConsent(RegisteredClient registeredClient,
```

Comment:
We should keep this method and associated logic as-is but change the signature to:

```java
private boolean requireAuthorizationConsent(OAuth2AuthorizationCodeRequestAuthenticationContext context)
```

We would also need to supply `OAuth2AuthorizationCodeRequestAuthenticationContext` with `OAuth2AuthorizationRequest` and `OAuth2AuthorizationConsent`.

This method would then be the default for:

```java
private Predicate<OAuth2AuthorizationCodeRequestAuthenticationContext> requiresAuthorizationConsent;
```

Applications could then override this via `setRequiresAuthorizationConsent()`.

Reply:
Sounds good. 

Would you like if `OAuth2AuthorizationRequest` and `OAuth2AuthorizationConsent` got treated the same way as `RegisteredClient` by exposing `getRegisteredClient()` and a builder method `registeredClient(RegisteredClient registeredClient)` inside the `OAuth2AuthorizationCodeRequestAuthenticationContext`? 

Or do you want just to use regular `.put()` and `.get()` for their supply & access?

Reply:
I have pushed a commit with a refactor according to your comment.

I attempted to expose getter & builders methods in the `OAuth2AuthorizationCodeRequestAuthenticationContext` for `OAuth2AuthorizationRequest` and `OAuth2AuthorizationConsent` instead of the `.put()` and `.get()` approach. However that ended up breaking a lot of tests. If It's important to go in that direction, I will put some more effort into making that work.